mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -640,6 +642,13 @@ convertOmpOrdered(Operation &opInst, llvm::IRBuilderBase &builder,
   return success();
 }
 
+static LogicalResult orderedRegionSupported(omp::OrderedRegionOp op) {
```

Comment:
Thanks for the comment @Meinersbur. I was actually planning on creating a follow-up PR to centralize this a bit more, but I decided that it might help address your comments so I made these changes here. Yes, the idea is to have this be called for all operations regardless of how many unimplemented clauses they have, for consistency. Hopefully the template approach I updated this PR with works for you.

Reply:
This should be ready to review again. Now the implementation is much more centralized, helping us make sure error messages are consistent.